crates/cache-key/src/canonical_url.rs

```diff
@@ -107,10 +107,13 @@ impl RepositoryUrl {
     pub fn new(url: &Url) -> RepositoryUrl {
         let mut url = CanonicalUrl::new(url).0;
 
+        // Clone to avoid borrow checker issues
+        let immutable_url = url.clone();
+
         // If a Git URL ends in a reference (like a branch, tag, or commit), remove it.
         if url.scheme().starts_with("git+") {
-            if let Some((prefix, _)) = url.as_str().rsplit_once('@') {
-                url = prefix.parse().unwrap();
+            if let Some((prefix, _)) = immutable_url.path().rsplit_once('@') {
+                url.set_path(prefix);
             }
         }
```

Comment:
This is awkward, is there a better pattern?

Reply:
I'll play with it, I'm not sure...

Reply:
One option:

```rust
if url.scheme().starts_with("git+") {
    if let Some(prefix) = url.path().rsplit_once('@').map(|(prefix, _suffix)| prefix.to_string()) {
        url.set_path(&prefix);
    }
}
```

Reply:
Still kind of strange but it seems better.

Reply:
Thanks!

Reply:
Like... mildly better because it doesn't clone the suffix, I guess, and it's contained without another binding.

Reply:
In the other case we need to clone the suffix too lol

Reply:
https://github.com/astral-sh/uv/pull/1717/commits/ae3ba9df5a6f13b8c291387547a9f2f3dfa8cc8e

Reply:
The cloned original URL bit is weird though I prefer this